extensions/amp-form/0.1/amp-form.js

```diff
@@ -488,14 +489,56 @@ export class AmpForm {
       dev().error(TAG, 'Form submit service failed: %s', e);
     }
 
+    // Get our special fields
     const varSubsFields = this.getVarSubsFields_();
-    if (this.xhrAction_) {
-      this.handleXhrSubmit_(varSubsFields, trust);
-    } else if (this.method_ == 'POST') {
-      this.handleNonXhrPost_();
-    } else if (this.method_ == 'GET') {
-      this.handleNonXhrGet_(varSubsFields);
+    const asyncInputs = toArray(
+        this.form_.querySelectorAll('.i-amphtml-async-input')
+    );
+
+    // Do any assertions we may need to do
+    // For NonXhrGET
+    // took out all the code in handleNonXhrGet_
+    // because alot was assertions we want to fail on before getting var subs
+    if (!this.xhrAction_ && this.method_ == 'GET') {
+      this.assertSsrTemplate_(false, 'Non-XHR GETs not supported.');
+      this.assertNoSensitiveFields_();
+
+      // If we have no async inputs, we can just submit synchronously
+      if (asyncInputs.length <= 0) {
```

Comment:
style nit: Will the length ever be less than `0`? You can use `==` to say exactly what you mean